cmd/libs/go2idl/client-gen/generators/client-generator.go

```diff
@@ -89,6 +90,18 @@ func packageForGroup(group string, version string, typeList []*types.Type, baseP
 				types:         typeList,
 				imports:       generator.NewImportTracker(),
 			})
+
+			expansionFileName := "generated_expansion"
+			// To avoid overriding user's manual modification, only generate the expansion file if it doesn't exist.
+			if _, err := os.Stat(filepath.Join(srcTreePath, outputPackagePath, expansionFileName+".go")); os.IsNotExist(err) {
+				generators = append(generators, &genExpansion{
+					DefaultGen: generator.DefaultGen{
+						OptionalName: expansionFileName,
+					},
+					types: typeList,
+				})
+			}
+
```

Comment:
While I understand why this is needed, this could also introduce an issue when a new resource is added, the expansion interface is embedded, but this file is not updated to define it. Not much you can do about it, just one of the many drawbacks of this approach.